google-ads-examples/src/main/java/com/google/ads/googleads/examples/extensions/AddAffiliateLocationExtensions.java

```diff
@@ -0,0 +1,397 @@
+ * a campaign for serving affiliate location extensions.
+ */
+public class AddAffiliateLocationExtensions {
+
+  private static class AddAffiliateLocationExtensionsParams extends CodeSampleParams {
+
+    @Parameter(names = ArgumentNames.CUSTOMER_ID, required = true)
+    private long customerId;
+
+    @Parameter(names = ArgumentNames.CAMPAIGN_ID, required = true)
+    private long campaignId;
+
+    @Parameter(
+        names = ArgumentNames.CHAIN_ID,
+        description =
+            "The retail chain ID. See"
+                + " https://developers.google.com/adwords/api/docs/appendix/codes-formats#chain-idsfor"
+                + " a complete list of valid retail chain IDs",
+        required = true)
+    private long chainId;
+  }
+
+  private static final int MAX_FEEDMAPPING_RETRIEVAL_ATTEMPTS = 10;
+
+  public static void main(String[] args) throws IOException {
+    AddAffiliateLocationExtensionsParams params = new AddAffiliateLocationExtensionsParams();
+    if (!params.parseArguments(args)) {
+
+      // Either pass the required parameters for this example on the command line, or insert them
+      // into the code here. See the parameter class definition above for descriptions.
+      params.customerId = Long.parseLong("INSERT_CUSTOMER_ID_HERE");
+    }
+
+    GoogleAdsClient googleAdsClient = null;
+    try {
+      googleAdsClient = GoogleAdsClient.newBuilder().fromPropertiesFile().build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      System.exit(1);
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      System.exit(1);
+    }
+
+    try {
+      new AddAffiliateLocationExtensions()
+          .runExample(googleAdsClient, params.customerId, params.campaignId, params.chainId);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      gae.printStackTrace();
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /** Runs the example. */
+  private void runExample(
+      GoogleAdsClient googleAdsClient, long customerId, long campaignId, long chainId) {
+    String feedResourceName =
+        createAffiliateLocationExtensionFeed(googleAdsClient, customerId, chainId);
+    // After the completion of the feed creation operation above the added feed will not
```

Comment:
Suggested edit (assuming I'm understand the flow correctly here): The feed created above will not be available for use in a campaign feed until the associated feed mappings have been created by Google. 
Attempts to retrieve the feedMapping using an exponential back-off policy until the feed mappings are created.

Reply:
Updated PTAL